diff -up wine-experimental_11.0/dlls/user32/tests/win.c.omv~ wine-experimental_11.0/dlls/user32/tests/win.c
--- wine-experimental_11.0/dlls/user32/tests/win.c.omv~	2024-09-27 21:48:59.037667156 +0200
+++ wine-experimental_11.0/dlls/user32/tests/win.c	2024-09-27 21:49:03.985659018 +0200
@@ -13226,5 +13226,5 @@ START_TEST(win)
     test_topmost();
 
     test_shell_window();
-    test_shell_tray();
+    //test_shell_tray();
 }
