diff -up wine-experimental_9.0/dlls/ntdll/env.c.omv~ wine-experimental_9.0/dlls/ntdll/env.c
diff -up wine-experimental_9.0/dlls/ntdll/tests/Makefile.in.omv~ wine-experimental_9.0/dlls/ntdll/tests/Makefile.in
diff -up wine-experimental_9.0/dlls/ntdll/tests/exception.c.omv~ wine-experimental_9.0/dlls/ntdll/tests/exception.c
--- wine-experimental_9.0/dlls/ntdll/tests/exception.c.omv~	2024-05-21 16:27:12.855868096 +0200
+++ wine-experimental_9.0/dlls/ntdll/tests/exception.c	2024-05-21 16:29:39.549136146 +0200
@@ -215,6 +215,7 @@ enum debugger_stages
 static int      my_argc;
 static char**   my_argv;
 static BOOL     is_wow64;
+static BOOL	old_wow64;
 static BOOL have_vectored_api;
 static enum debugger_stages test_stage;
 
@@ -12701,6 +12702,17 @@ START_TEST(exception)
     X(IsWow64Process);
     if (!pIsWow64Process || !pIsWow64Process( GetCurrentProcess(), &is_wow64 )) is_wow64 = FALSE;
 
+    if (is_wow64)
+    {
+        TEB64 *teb64 = ULongToPtr( NtCurrentTeb()->GdiBatchCount );
+
+        if (teb64)
+        {
+            PEB64 *peb64 = ULongToPtr(teb64->Peb);
+            old_wow64 = !peb64->LdrData;
+        }
+    }
+
     X(InitializeContext);
     X(InitializeContext2);
     X(LocateXStateFeature);
diff -up wine-experimental_9.0/dlls/ntdll/tests/info.c.omv~ wine-experimental_9.0/dlls/ntdll/tests/info.c
diff -up wine-experimental_9.0/dlls/ntdll/unix/env.c.omv~ wine-experimental_9.0/dlls/ntdll/unix/env.c
--- wine-experimental_9.0/dlls/ntdll/unix/env.c.omv~	2024-05-21 16:20:28.464567681 +0200
+++ wine-experimental_9.0/dlls/ntdll/unix/env.c	2024-05-21 16:20:38.392643877 +0200
@@ -62,7 +62,7 @@ WINE_DEFAULT_DEBUG_CHANNEL(environ);
 WINE_DECLARE_DEBUG_CHANNEL(nls);
 
 PEB *peb = NULL;
-WOW_PEB *wow_peb = NULL;
+__attribute__((used)) WOW_PEB *wow_peb = NULL;
 USHORT *uctable = NULL, *lctable = NULL;
 SIZE_T startup_info_size = 0;
 BOOL is_prefix_bootstrap = FALSE;
