--- wine-experimental_11.0/dlls/ntdll/unix/signal_x86_64.c.omv~	2026-07-25 22:31:17.854430238 +0200
+++ wine-experimental_11.0/dlls/ntdll/unix/signal_x86_64.c	2026-07-25 22:32:33.116734052 +0200
@@ -2279,7 +2279,7 @@ __ASM_GLOBAL_FUNC( dump_syscall_fault_re
                    "movq %rdi,%rsp\n\t"
                    "movq %rsi,%rax\n\t"
                    "movq %rdx,%r13\n\t"
-                   "jmp %rcx")
+                   "jmp *%rcx")
 
 
 static void dump_syscall_fault( CONTEXT *context, DWORD exc_code )
