diff -up wine-experimental_11.0/dlls/winevulkan/Makefile.in.1~ wine-experimental_11.0/dlls/winevulkan/Makefile.in
--- wine-experimental_11.0/dlls/winevulkan/Makefile.in.1~	2026-06-18 15:56:22.000000000 +0200
+++ wine-experimental_11.0/dlls/winevulkan/Makefile.in	2026-07-25 22:15:28.810571665 +0200
@@ -2,7 +2,7 @@ MODULE    = winevulkan.dll
 UNIXLIB   = winevulkan.so
 IMPORTLIB = winevulkan
 IMPORTS   = advapi32
-UNIX_LIBS = -lwin32u $(PTHREAD_LIBS)
+UNIX_LIBS = -lwin32u -lm $(PTHREAD_LIBS)
 
 SOURCES = \
 	loader.c \
